### PR TITLE
Crm 17422 fix

### PR DESCRIPTION
**Pull request by monishdeb**

### CODE REVIEW

File: CRM/Core/BAO/CustomValue.php [hunk 1]
```diff
@@ -189,11 +189,11 @@ public static function fixCustomFieldValue(&$formValues) {
       elseif (($htmlType == 'TextArea' ||
           ($htmlType == 'Text' && $dataType == 'String')
         ) &&
-        !((substr($formValues[$key], 0, 1) == '%') ||
```
Comment (eileenmcnaughton) [inline CRM/Core/BAO/CustomValue.php]: That fixes my symptom - but I think this is better
``
      elseif (stristr($formValues[$key], '%') && ($htmlType == 'TextArea' ||
         ($htmlType == 'Text' && $dataType == 'String')	          ($htmlType == 'Text' && $dataType == 'String')

``

**Comment by eileenmcnaughton** [comment]
I tested this & it worked for me - I tested on custom data.

A couple of notes though
1) if the behaviour changed in ?4.6 ? 4.5? & then is changing back in 4.7 I'm inclined to put this to 4.6 - so that people don't see the behaviour change too often - if they actually notice.

2) I'm trying to figure out how to add a test for this. I can see that it's the sort of thing with no impact on a small DB that devs are likely to be on. Having a change to the query cause a well commented test to fail seems like the best protection

**Comment by monishdeb** [comment]
I was confused with this condition 
`` (!((substr($formValues[$key], 0, 1) == '%') ||(substr($formValues[$key], -1, 1) == '%')) ``
as a text may contain wildcard character as a part of the string e.g. - '...10% cut of share...' So in order to differentiate this two use cases
 1. Presence of willdcard as an part of the text
 2. User wants to use wildcard for custom text search
 Reason why they might have used that condition earlier (prior to 4.4) ONLY to check the presense of wildcard at start/end not as if it contains or not.

**Comment by eileenmcnaughton** [comment]
Yeah 10% is pretty odd! OTOH it would be rare but not unheard of to want to search for An%e to catch more than one variant.

I agree both versions are a bit of an edge case so I'm not strongly advocating for it. Arguably our customers might use the more flexible search if I told them it existing.

**Comment by monishdeb** [comment]
As an alternative if a user need to search with actual wildcard charater then he need to escape it ( .... \% cut of share....) in advance search.

**Comment by eileenmcnaughton** [comment]
yeah - not realistic from a training point of view. We do teach our customers somewhat regularly about putting just % in a field - e.g to find has an email address

**Comment by monishdeb** [comment]
Jenkin, test this please.

**Pull request merged**